miners/IMAGE/skeleton/skeleton.py

```diff
@@ -0,0 +1,250 @@
+                    self.best_test_loss = test_loss # Update best loss.
+                    self.model_toolbox.save_model(
+                        self.config.miner.full_path,
+                        {
+                            'epoch': self.epoch, 
+                            'model_state_dict': self.model.state_dict(), 
+                            'loss': self.best_test_loss,
+                            'optimizer_state_dict': self.optimizer.state_dict(),
+                        }
+                    )
+                    self.tensorboard.add_scalar('Test loss', test_loss, self.global_step)
+
+    # ---- Train epoch ----
+    def train(self):
+        # ---- Init training state ----
+        self.model.train() # Turn on dropout etc.
+        for batch_idx, (images, targets) in enumerate(self.trainloader): 
+            if batch_idx >= self.config.miner.epoch_length:
+                break   
+            self.global_step += 1 
+
+            # ---- Remote Forward pass ----
+            output = self.model.remote_forward(  
+                neuron = self.neuron,
+                images = images.to(self.device), 
+                targets = torch.LongTensor(targets).to(self.device), 
+            ) 
+            
+            # ---- Remote Backward pass ----
+            loss = output.remote_target_loss + output.local_target_loss + output.distillation_loss
+            loss.backward() # Accumulates gradients on the model.
+            clip_grad_norm_(self.model.parameters(), self.config.miner.clip_gradients) # clip model gradients
+            self.optimizer.step() # Applies accumulated gradients.
+            self.optimizer.zero_grad() # Zeros out gradients for next accummulation 
+
+            # ---- Train weights ----
+            batch_weights = torch.mean(output.router.weights, axis = 0).to(self.model.device) # Average over batch.
+            self.row = (1 - 0.03) * self.row + 0.03 * batch_weights # Moving avg update.
+            self.row = F.normalize(self.row, p = 1, dim = 0) # Ensure normalization.
+
+            # ---- Step Logs + Tensorboard ----
+            processed = ((batch_idx + 1) * self.config.miner.batch_size_train)
+            progress = (100. * processed) / len(self.train_data)
+            logger.info('GS: {}\t Epoch: {} [{}/{} ({})]\tLoss: {}\tAcc: {}\tAxon: {}\tDendrite: {}', 
+                    colored('{}'.format(self.global_step), 'blue'), 
+                    colored('{}'.format(self.epoch), 'blue'), 
+                    colored('{}'.format(processed), 'green'), 
+                    colored('{}'.format(len(self.train_data)), 'red'),
+                    colored('{:.2f}%'.format(progress), 'green'),
+                    colored('{:.4f}'.format(output.local_target_loss.item()), 'green'),
+                    colored('{:.4f}'.format(output.local_accuracy.item()), 'green'),
+                    self.neuron.axon,
+                    self.neuron.dendrite)
```

Comment:
Can you change this to using qqdm instead? it will print logs in place instead of new lines. refer to: https://github.com/opentensor/bittensor/pull/253